src/Function.agda

```diff
@@ -100,6 +100,16 @@ case_return_of_ : ∀ {A : Set a} (x : A) (B : A → Set b) →
                   ((x : A) → B x) → B x
 case x return B of f = f x
 
+-- Converting between implicit and explicit function spaces.
+
+_$- : ∀ {A : Set a}{B : A → Set b} →
+      (∀ x → B x) → (∀ {x} → B x)
+f $- = f _
+
+_$· : ∀ {A : Set a}{B : A → Set b} →
+       ({x : A} → B x) → ((x : A) → B x)
+f $· = λ x → f
+
```

Comment:
Should this be called `λ-_` instead? This way `(λ- f) $- ≡ f`

Reply:
And also `λ- (f $-) ≡ f` aka eta-contraction.

Reply:
Good suggestion! You are clearly much better than me at this naming business...